### ISSUE TITLE
Fix tests for pandas 0.24

### ISSUE THREAD

**Issue by flying-sheep** [theislab/anndata#105]
The tests in PR #103 ran through, but after that and before merging, pandas 0.24 was released with some changes to categorical arrays that broke our tests. I fixed one of the consequences in e127e6ce79299b4a47f47b7ea01c7daf6ea85342 but more needs to be done.

see also theislab/scanpy#448

**Comment by flying-sheep** [comment]
There were some wrong assumptions in the tests. The following actually succeeds:

```py
assert pd.api.types.is_string_dtype(pd.Categorical(['a', 'b']))
```

What we wanted to test is simply that some annotation is not categorical:

```py
assert not pd.api.types.is_categorical(adata.obs['something'])
```

**Comment by flying-sheep** [comment]
Fixed in 4c8163129302391419c7ee4943e7fb766599e2a2 and friends
